gslib/context_config.py

```diff
@@ -107,6 +113,62 @@ def _SplitPemIntoSections(contents, logger):
   return result
 
 
+def _check_path():
+  """Checks for content aware metadata.
+
+  If content aware metadata exists, return its absolute path;
+  otherwise, returns None.
+
+  Returns:
+    str: Absolute path if exists. Otherwise, None.
+  """
+  metadata_path = os.path.expanduser(_DEFAULT_METADATA_PATH)
+  if not os.path.exists(metadata_path):
+    return None
+  return metadata_path
+
+
+def _read_metadata_file(metadata_path):
+  """Loads context aware metadata from the given path.
+
+  Returns:
+      Dict[str, str]: The metadata.
+
+  Raises:
+      CertProvisionError: If failed to parse metadata as JSON.
+  """
+  try:
+    with open(metadata_path) as f:
+      return json.load(f)
+  except ValueError as e:
+    raise CertProvisionError(e)
+
+
+def _default_command():
+  """Loads default cert provider command.
+
+  Returns:
+      str: The default command.
+
+  Raises:
+      CertProvisionError: If command cannot be found.
+  """
+  metadata_path = _check_path()
+
+  if not metadata_path:
+    raise CertProvisionError("Client certificate provider is not found.")
```

Comment:
I would write this as "Client certificate provider file not found."

Then, you can write the next `CertProvisionError` as "Client certificate provider  command not found."